cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -780,11 +781,20 @@ impl Inst {
         }
     }
 
-    pub(crate) fn xmm_rm_r_imm(op: SseOpcode, src: RegMem, dst: Writable<Reg>, imm: u8) -> Inst {
-        src.assert_regclass_is(RegClass::V128);
-        debug_assert!(dst.to_reg().get_class() == RegClass::V128);
-        debug_assert!(imm < 8);
```

Comment:
These assertions were only valid for `CMPP*`; with the addition of `PINSR*` and `PEXTR*` the reg classes can come from I64 in either slot (perhaps we could keep the `imm` assertion with `imm < 16`).